src/lib/Libattr/attr_fn_l.c

```diff
@@ -206,15 +206,58 @@ set_l(struct attribute *attr, struct attribute *new, enum batch_op op)
 
 /**
  * @brief
- *	comp_l - compare two attributes of type long
+ * 	set_l_notneg - Set a long but don't allow negative numbers.
  *
- * @param[in] attr - pointer to attribute structure
- * @param[in] with - pointer to attribute structure
+ * @param[in/out]   attr - pointer to new attribute to be set (A)
+ * @param[in]   new  - pointer to attribute (B)
+ * @param[in]   op   - operator
  *
  * @return      int
- * @retval      0       if the set of strings in "with" is a subset of "attr"
- * @retval      1       otherwise
+ * @retval      0       if ok
+ * @retval     >0       if error
+ *
+ */
+int
+set_l_notneg(struct attribute *attr, struct attribute *new, enum batch_op op)
+{
+	long	val;
+
+	if (!(attr && new && (new->at_flags & ATR_VFLAG_SET)))
```

Comment:
U can move this check code  action function.  set_l_notneg is not necessary

Reply:
You can user set_l() as usually and do the rest of the validation that is done here in the corresponding action function.

Reply:
What about using the ecl library?  Use the verify_value_zero_or_positive() validation function.

Reply:
Fixed. Used "zero_or_positive_action" instead along with set_l.

@bhroam Since these attributes get set through hooks I am afraid ecl wouldn't be of much help.